submit/filters/tex_filters.py

```diff
@@ -0,0 +1,258 @@
+"""Filters for highlighting autotex log files."""
+
+import re
+
+
+TEX = 'tex'
+LATEX = 'latex'
+PDFLATEX = 'pdflatex'
+
+ENABLE_TEX = r'(\~+\sRunning tex.*\s\~+)'
+ENABLE_LATEX = r'(\~+\sRunning latex.*\s\~+)'
+ENABLE_PDFLATEX = r'(\~+\sRunning pdflatex.*\s\~+)'
+
+DISABLE_HTEX = r'(\~+\sRunning htex.*\s\~+)'
+DISABLE_HLATEX = r'(\~+\sRunning hlatex.*\s\~+)'
+DISABLE_HPDFLATEX = r'(\~+\sRunning hpdflatex.*\s\~+)'
+
+RUN_ORDER = ['last', 'first', 'second', 'third', 'forth']
+
+def compilation_log_display(autotex_log: str) -> str:
```

Comment:
For testing, I would focus on examples. Here is a similar situation: https://github.com/arXiv/arxiv-base/blob/master/arxiv/base/urls/tests/test_urlize.py

So if you have a sample line for each of the patterns you are testing, you would be asserting that the expected HTML-annotated output is produced

Reply:
So basically test at the routine level, pass in a controlled log and assert the desired outcome. I will try to add something shortly.

Reply:
Yep; and if possible using short examples -- easier to see what is being tested